### DISCUSSION TITLE
Advice on best performance. AI analysis and transcript to SRT.

### DISCUSSION THREAD

**Discussion by FabriceACohen** [deepgram/community#192] [General help] [answered]
Hello,
I am new to deepgram.

What would be your advice for using Deepgram to batch transcript audios of vidéos with :
- creation of srt file for beeing able to add captions to the video.
- AI analysis : language, topic detection, ... to add to a database.

What language ? Python, curl, ... I struggled trying to create an srt file in python for a batch transcript request.

Many thanks.
Fabrice.

**Comment by jkroll-deepgram** [comment]
Hi Fabrice, currently we do offer a `.toSRT()` method within the [Node SDK](https://developers.deepgram.com/docs/node-sdk-pre-recorded-transcription#converting-to-srt). We also have a [blog post ](https://blog.deepgram.com/generate-webvtt-srt-captions-nodejs/) about creating that feature. We don't offer out-of-the-box support in Python yet, but hopefully the Node examples provide some guidance for you. Meanwhile, I will look at reaching parity across our SDKs to support SRT file creation within the Python SDK.

Deepgram does support both [language detection](https://developers.deepgram.com/docs/language-detection) and [topic detection](https://developers.deepgram.com/docs/topic-detection), so you could get that information from your audio files using Deepgram, and then write it to your database.

**Reply by FabriceACohen** [reply]
Thank you very much. Can I ask you another question.
During my tests I couldn't understand the billing with my parameters. The princing page shows all model around $0.004/minute

My log shows : Callback URL
Audio duration 00:00:53
Cost $0.01305
Model 6e79c099-c46b-40f1-82eb-1656e5196236
Method Synchronous
Language fr
Tier -
Punctuate True

Can you explain what I am doing wrong ?

**Reply by jjmaldonis** [reply]
Hi Fabrice, that is a bit strange. Were there additional parameters included in the request, such as Summarization, Topic Detection, or Entity Detection? Those features cost extra, so that could be what happened. If you can find the `Request ID` in your logs and copy-paste it here, we can look into the details.

**Reply by FabriceACohen** [reply]
Thank you. Here are a few Request IDs I don't understand. Even the ones with extra cost features, I don't understand the splitting and the sum of the cost :
2e58f598-4e4e-40bb-b1b7-4033de89861d
36a4942a-6bd6-4bd2-ac78-92150be75ae4
a3846e35-6743-47e3-b7d4-b4fadfb798f8
173a9e55-e5fd-4468-9277-7f2b926fc0af

**Reply by jjmaldonis** [reply, marked as answer]
Hey @FabriceACohen, it turns out the cost difference you're seeing is because the `enhanced` model is being used rather than `nova`. The enhanced model is ~3x more expensive, which is where the additional cost is coming from.

If you're able to use either the `nova` model or the `base` model, it will be noticeably less expensive. The enhanced models are particularly powerful for transcribing meetings and phone calls, or when financial terms are used in the conversation.
